### DISCUSSION TITLE
B: Fuzziness: missing data and impedance mismatch

### DISCUSSION THREAD

**Discussion by ml-evs** [MADICES/MADICES-2022#4] [Ideas]
This thread is for discussing, finalizing and organizing a breakout with the theme in the title above. 

[Link to current description](https://github.com/MADICES/MADICES-2022/blob/main/breakouts/B_fuzziness.md)

> # Fuzziness: dealing with missing data and impedance mismatch
> - How do we deal with details that are missing in the experimental data but are key parts in e.g. computational data or for ML?  
> - Most ELNs do not even record structured/FAIR-compliant data. How do we bring them into the loop? 
> - Is there a need/place for a “FAIRification” web service? (This goes a bit along the lines of Damien’s “Archive forge”, or the frictionless data interface)
> - What are the basic abstractions to characterize an experimental sample? 
>     - Is it a “sample”, a “compound” and is a “compound” a mixture of “chemicals” or something completely different? 
> - How are links between notebook “entries” (e.g. samples, devices, precursors, measurements, experiments, projects) represented?

**Comment by kjappelbaum** [comment]
closely related to fuzziness is also usability: You might have a long form asking for a lot of details --- but no one will use this. 
How can one hide a lot of the metadata capture from the user (part of the answer will probably be lab automation). 

also related is uncertainty/disorder: experimental (crystal) structures are often "disordered". On the one hand you will always have an ensemble of different structures (+impurities) in your sample and, on the other hand, XRD can often not resolve the position of all atoms. For latter there is somewhat of a standard in CIF, however for the former it is not really clear.

**Reply by sphuber** [reply]
> You might have a long form asking for a lot of details --- but no one will use this.

This is a very good point and one that came up in a discussion I had recently. I think it is true indeed that it is unlikely that putting the burden on the experimentalist of entering data in structured format will result in more structured data. The focus should instead be to automate as much as possible, and that whatever data is entered by humans can still be obtained programmatically which is then analyzed with NLP algorithms, which are becoming ever more powerful, to structure the data after the fact.

**Reply by petermr** [reply]
NLP
I am a believer in implementing how people currently think, rather than the hypothetical future. CML was modelled by taking existing discourse and working out what was important in it. The basic narrative has been the same for 150 years:

```
for all X, Y, Z  C in our area of interest:
    add X to Y w conditions C to create Z, purify Z and measure its properties
    change and repeat
```
So Lezan Hawizy in our group created http://chemicaltagger.ch.cam.ac.uk/ which applies rules and NLP to chemical syntheses. If someone has time the properties, procedures, solvents, etc could be linked to Wikidata. NLP is a lot easier now than 10 years ago.

If you use current language and apply NLP you'll end up with something that people relate to.

**Comment by giovannipizzi** [comment]
I have another topic that is maybe partially unrelated to the discussions above, but I think still fits in this discussion topic (and partially mentioned above "What are the basic abstractions to characterize an experimental sample?").

* In a simulation, an input is e.g. a crystal structure. This is an immutable concept. I can have a calculation that relaxes the atom coordinates, but I can imagine this as a simulation generating a new crystal structure, different from the previous one: I give to the relaxed structure a different ID and it's a different piece of data by itself (and can be an input to another simulation). In addition, I can perform any type of simulations on the initial crystal structure, and I'm never going to modify or "destroy" it

* In experiments, the inputs are samples. However, some measurements can be destructive (or more generally they can change the history of the sample, so the sample after the measurement is different than before, and can e.g. yield different results in subsequent experiments). 

Here are the questions I have:

* How to match the two worlds? How to represent, in a workflow tool to manage simulations, the concept of an experimental sample and its state, that maybe does not exist anymore (because the sample is there, but some measurements have been performed on it that changed its state)?

* How to represent in an experimental ELN the fact that I performed a simulation on material "X" (say, silicon), where I could have many silicon samples, all different, but that should all be represented by the same simulation on crystalline silicon?

That is to say:
- I have a many-to-many relation between the inputs in simulations and in experiments (I can have multiple Si ideal crystal structures (SIMULATION), representing multiple domains of the same sample of polycrystalline Si (EXPERIMENT); or I can have multiple Si mono crystalline samples (EXPERIMENT) all represented by the same Si crystal structure (SIMULATION).

- the problem can probably find a not-too-complex solution in the case where we want simply to represent the simulation results in an experimental ELN (they can become a linked annotation to a number of Si sample entries etc.). Or if we want to keep track of a given sample with its state in a simulation, to say "it's a simulation for that state of the sample, e.g. because it later changed crystalline state after melting" (by storing in some way the state of the sample together with the ID - in the most naive approach, e.g. by storing a time; one can then go in the experimental ELN to check, at that time, what was the state of the sample). These solutions, by the way, require a way for the workflow engine and the experimental ELN to exchange data in some format (and there is thus the problem of impedance mismatch we are discussiong).
 
- However, these questions come from a current collaboration where we want to drive automatic robotic experiments (e.g. to build a coin-cell battery), with experiments driven by a workflow manager. Therefore, because of the automation, the two words need to speak together: I want to use concepts of simulations/workflow engines to manage the experiments, but at the same time I'm dealing with inputs and outputs that are in the experimental world (samples, that are not immutable but whose history matters).

**Comment by kjappelbaum** [comment]
ulrich from chat: 

> I think this is one of the core problems - we see a lot of nice talks here About vocabularies, ontologies, text mining - but chemistry in inherently "fuzzy"
For example, I can easily see that "trepene" should really be "terpene" if I have the rest of a paper for conext but again have the Computer do that

**Reply by kjappelbaum** [reply]
Kenneth from chat: 

>Reaction to ^^ and question for the current speaker: it’s not so much that the language is fuzzy, but that what is sufficient specificity in one context is insufficient in another.  My fancy way of expressing that is “Jargon is nested fractally.”  How do you address cross-domain applications given this?

**Comment by rlaplaza** [comment]
IMHO chemistry at least is far too fuzzy for any "cathedral" model. We are, to a degree, pushed towards a decentralized model of coexisting standards, hopefully interoperable through painless parsing. Every micro-community will have a particular ontology and a way to define their connections in the form of metadata, which will not be transferable to closely related communities (i.e. is enantioselectivity a reaction property or a product/compound property? depends on who you ask).

**Comment by petermr** [comment]
In CML we had the following "objects":
* molecule. This was an abstraction and could cover connection tables (an
abstraction),
" formula". formulae of various sorts, not necessarily integral. They can
contain free variables (x, y, etc.). (Many CML objects can have free
variables - it's very powerful, and the free variables can have ranges and
other constraints).
* crystal. An idealised 2- or 3-D lattice, usually infinite but could be
limited
* substance. This represents the real world and covers mixtures, samples,
batches, etc. It accepts the fuzziness which is described by annotation.
* transform3 and transform2 (allows any object with coordinates to be
replicated in different positions).


On Tue, Feb 8, 2022 at 9:19 AM Rubén Laplaza ***@***.***>
wrote:

> IMHO chemistry at least is far too fuzzy for any "cathedral" model. We
> are, to a degree, pushed towards a decentralized model of coexisting
> standards, hopefully interoperable through painless parsing. Every
> micro-community will have a particular ontology and a way to define their
> connections in the form of metadata, which will not be transferable to
> closely related communities (i.e. is enantioselectivity a reaction property
> or a product/compound property? depends on who you ask).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MADICES/MADICES-2022/discussions/4#discussioncomment-2131774>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS5V7D6MKFSM4SNE7YTU2DN2FANCNFSM5KQPFG6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK
